src/Chapter3.hs

```diff
@@ -971,6 +1071,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+{- ORMOLU_ENABLE -}
+data Weekday = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Show, Eq, Ord, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend x
+  | x == Saturday || x == Sunday = True
+  | otherwise = False
+
+nextDay :: Weekday -> Weekday
+nextDay Sunday = Monday
+nextDay day = succ day
+
+daysToParty :: Weekday -> Int
+daysToParty day =
+  length
+    ( takeWhile
+        (/= Friday)
+        ( drop
+            (fromEnum day)
+            (cycle (enumFrom Monday))
+        )
+    )
```

Comment:
Interesting.. But it is far from an optimal solution as you can guess 😄 
 
You could just use the `Enum` instance here and the fact that there is 7 days in a weak. You can find `mod ... 7` useful 🙂